indexer-api/src/infra/api/v1/subscription/unshielded.rs

```diff
@@ -80,114 +83,177 @@ where
+                            .get_highest_indices_for_address(&address)
+                            .await
+                            .internal("fetch highest indices for address")?;
+                        let highest_transaction_id = highest_transaction_id.unwrap_or(0);
+
+                        let progress = UnshieldedProgress {
+                            highest_transaction_id,
+                            current_transaction_id: transaction.id,
+                        };
+
+                        let created_utxos = Some(created.into_iter()
+                            .map(|utxo| UnshieldedUtxo::<S>::from((utxo, network_id)))
+                            .collect());
+                        let spent_utxos = Some(spent.into_iter()
+                            .map(|utxo| UnshieldedUtxo::<S>::from((utxo, network_id)))
+                            .collect());
+
+                        yield UnshieldedUtxoEvent {
+                            progress,
+                            transaction: Some(transaction.into()),
+                            created_utxos,
+                            spent_utxos,
+                        };
+                    }
                 }
-            };
-
-            let created = storage
-                .get_unshielded_utxos_created_in_transaction_for_address(&address, transaction_id)
-                .await
-                .internal("fetch created UTXOs")?;
-
-            let spent = storage
-                .get_unshielded_utxos_spent_in_transaction_for_address(&address, transaction_id)
-                .await
-                .internal("fetch spent UTXOs")?;
-
-            let (highest_index, _) = storage
-                .get_highest_indices_for_address(&address)
-                .await
-                .internal("fetch highest indices for address")?
-                ;
-            let highest_index = highest_index.unwrap_or(0);
-
-            let current_index = tx.end_index;
-
-            let progress = UnshieldedProgress {
-                highest_index,
-                current_index,
-            };
-
-            yield UnshieldedUtxoEvent {
-                progress,
-                transaction: Some(tx.into()),
-                created_utxos: Some(created.into_iter()
-                    .map(|utxo| UnshieldedUtxo::<S>::from((utxo, network_id)))
-                    .collect()),
-                spent_utxos: Some(spent.into_iter()
-                    .map(|utxo| UnshieldedUtxo::<S>::from((utxo, network_id)))
-                    .collect()),
-            };
-        }
 
-        warn!("stream of UnshieldedUtxoIndexed events completed unexpectedly");
-    };
+                // Phase 2: Stream live events, skipping any already processed.
+                let utxo_indexed_events = {
+                    let address = address.clone();
 
-    Ok(updates)
+                    utxo_indexed_events
+                        .try_filter(move |event| ready(event.address == address))
```

Comment:
Why not invoke `try_filter` above when defining `utxo_indexed_events` (line 93)?